main/src/modules/contextualLayer/goalManager/src/goalManager.cpp

```diff
@@ -0,0 +1,216 @@
+#include "goalManager.h"
+#include "wrdac/subsystems/subSystem_recog.h"
+
+bool GoalManager::configure(yarp::os::ResourceFinder &rf)
+{
+    string moduleName = rf.check("name", Value("goalManager")).asString().c_str();
+    setName(moduleName.c_str());
+
+    yInfo() << moduleName << " : finding configuration files...";
+    period = rf.check("period", Value(0.1)).asDouble();
+
+    //Create an iCub Client and check that all dependencies are here before starting
+    bool isRFVerbose = false;
+    //iCub = new ICubClient(moduleName, "goalManager", "client.ini", isRFVerbose);
+    //iCub->opc->isVerbose = false;
+    //if (!iCub->connect())
+    //{
+    //    yInfo() << " iCubClient : Some dependencies are not running...";
+    //    Time::delay(1.0);
+    //}
+    rpc.open(("/" + moduleName + "/rpc").c_str());
+    attach(rpc);
```

Comment:
For good style, can you move the `attach` method to the end of `configure`? The RPC port should only be reachable if the module is configured.